### PR TITLE
use while instead of foreach and return

### PR DESCRIPTION
**Pull request by xuwei-k**
avoid NonLocalReturnControl

### CODE REVIEW

File: src/library/scala/runtime/Tuple2Zipped.scala [hunk 1]
```diff
@@ -44,62 +45,75 @@ final class Tuple2Zipped[El1, It1 <: Iterable[El1], El2, It2 <: Iterable[El2]](p
...
     b.sizeHint(coll1, 0)
+    val elems1 = coll1.iterator
     val elems2 = coll2.iterator
 
-    for (el1 <- coll1) {
-      if (elems2.hasNext)
-        b += f(el1, elems2.next())
-      else
-        return b.result()
+    def loop(): To = {
```
Comment (allanrenucci) [inline src/library/scala/runtime/Tuple2Zipped.scala]: Why not a while loop? It is more concise
```scala
while (elems1.hasNext && elems2.hasNext)
  b += f(elems1.next(), elems2.next())

b.result()
```

**Comment by szeiger** [comment]
Do you have any benchmarks for this change?

**Comment by lrytz** [comment]
I'd merge this one. @szeiger?

**Comment by adriaanm** [comment]
The while loops look cleaner to me, actually.

**Pull request merged**